### ISSUE TITLE
Redux dev tools one step behind.

### ISSUE THREAD

**Issue by tomitrescak** [mobxjs/mobx-state-tree#287]
Hi, thanks for all the amazing work! I have an issue with connecting Redux dev tools to mobx state tree. While the connection works like charm, it seem that the process is one step behind. Please check the GIF. When I switch my UI component from "signIn" view to "resend verification" it dev tools the view change is recorder as "verification -> sign in".

![mobx-state-tree](https://user-images.githubusercontent.com/2682705/28949402-f5496450-78ff-11e7-9893-7f0fd4370fc9.gif)

This is how I initialise my store:

```ts
const state = accountStore.create();
global.__store = state;
connectReduxDevtools(require('remotedev'), state);
state.setView('signIn');
return <AccountsRoot state={state} />;
```

**Comment by tinchoz49** [comment]
I had exactly the same issue today.

**Comment by theorye** [comment]
Are you guys using the remotedev that connects to the remotedev.io server or a local server?

**Comment by tinchoz49** [comment]
Yes, I'm using remotedev.io server. It's happening the same problem in the todomvc example. 

I think the problem is that redux for example runs a `@@INIT` action at the begin and then the next actions can do a diff of the state correctly.

In our case, MST doesn't have an init action and I don't think that is necessary, maybe we can simulate it to fix this issue.

**Comment by mweststrate** [comment]
Generating an `@@INIT` with the initial snapshot when setting up the
wrapper is trivial probably fixes it indeed

Op vr 11 aug. 2017 om 23:22 schreef Martín Acosta <notifications@github.com

**Comment by tinchoz49** [comment]
Hi @mweststrate @mattiamanzati! I've been thinking that maybe there is an issue here: https://github.com/mobxjs/mobx-state-tree/blob/80d98e41aa51914619532a28773cfaf41eada426/src/interop/redux.ts#L71

When you execute an action, MST detect the action using the `onAction`, a middleware that it's executing before the action applys the changes on the store, so when you get an snapshot of the store you are getting the old state and redux-devtools can't diff correctly.

Maybe we can add new middleware `onAfterAction` or I was thinking in change the original onAction to:

```typescript
onAction(
    target: IStateTreeNode,
    beforeListener: (call: ISerializedActionCall) => void, 
    afterListener: (call: ISerializedActionCall) => void 
)
```

What do you think?

**Comment by mattiamanzati** [comment]
Fixed in 0.10.3
